### PR TITLE
Added the rust-lang regex project.

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Cross-referencing [PR](https://github.com/rust-lang/regex/pull/679) 

@BurntSushi is one of the maintainers of rust-lang Regex and his email is in the project.yaml. 

The repository has more than 19 million downloads [here](https://crates.io/crates/regex) and more than 2000 projects depend on it shown [here](https://crates.io/crates/regex/reverse_dependencies)  

The library is in the top 15 most downloaded Rust projects [here](https://crates.io/crates?sort=downloads) 

Waiting for the merge upstream.

### CODE REVIEW

File: projects/rust-regex/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER david@adalogics.com
+RUN apt-get update && apt-get install -y make autoconf automake libtool curl cmake python llvm-dev libclang-dev clang
+
+ENV CARGO_HOME=/rust RUSTUP_HOME=/rust/rustup PATH=$PATH:/rust/bin
```
Comment (inferno-chromium) [inline projects/rust-regex/Dockerfile:20]: This is not needed now.

File: projects/rust-regex/build.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# Note: This project creates Rust fuzz targets exclusively
+export CUSTOM_LIBFUZZER_PATH="$LIB_FUZZING_ENGINE_DEPRECATED"
```
Comment (inferno-chromium) [inline projects/rust-regex/build.sh:19]: This is not needed now.

File: projects/rust-regex/build.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+export CUSTOM_LIBFUZZER_STD_CXX=c++
+
+# Because Rust does not support sanitizers via CFLAGS/CXXFLAGS, the environment
+# variables are overridden with values from base-images/base-clang only
+export CFLAGS="-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION"
```
Comment (inferno-chromium) [inline projects/rust-regex/build.sh:24]: This all is not needed now.

File: projects/rust-regex/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER david@adalogics.com
+RUN apt-get update && apt-get install -y make autoconf automake libtool curl cmake python llvm-dev libclang-dev clang
```
Comment (inferno-chromium) [inline projects/rust-regex/Dockerfile:18]: why install clang libclang-dev etc ? these should be there in image.

**Comment by inferno-chromium** [comment]
We accept this project, thanks for citing the rust dependencies. Can you fix the travis failures, then we can merge.

**Comment by DavidKorczynski** [comment]
Hey @inferno-chromium -  quick notification here that the fuzzers are now merged upstream(https://github.com/rust-lang/regex/pull/679) and Travis passes

**Review by inferno-chromium — APPROVED** [review]

**Pull request merged**